src/utils/filenamehandler.cpp

```diff
@@ -52,70 +52,71 @@ QString FileNameHandler::parseFilename(const QString& name)
     return res;
 }
 
-QString FileNameHandler::generateAbsolutePath(const QString& path)
+/**
+ * @brief Generate a valid destination path from the possibly incomplete `path`.
+ * The input `path` can be one of:
+ * - empty string
+ * - an existing directory
+ * - a file in an existing directory
+ * In each case, the output path will be an absolute path to a file with a
+ * suffix matching the specified `format`.
+ * @note
+ * - If `path` points to a directory, the file name will be generated from the
+ *   formatted file name from the user configuration.
+ * - If `path` points to a file, its suffix will be changed to match `format`
+ * - If `format` is not given, the suffix will remain untouched.
+ * - If the path generated by the previous steps points to an existing file,
+ * 	 "_NUM" will be appended to its base name, where NUM is the first
+ * available number that produces a non-existent path (starting from 1).
+ * @param path Possibly incomplete file name to transform
+ * @param format Desired output file suffix (excluding an initial '.' character)
+ */
+QString FileNameHandler::properScreenshotPath(QString path,
+                                              const QString& format)
 {
-    QString directory = path;
-    QString filename = parsedPattern();
-    fixPath(directory, filename);
-    return directory + filename;
-}
-// path a images si no existe, add numeration
-void FileNameHandler::setPattern(const QString& pattern)
-{
-    ConfigHandler().setFilenamePattern(pattern);
-}
-
-QString FileNameHandler::absoluteSavePath(QString& directory, QString& filename)
-{
-    ConfigHandler config;
-    directory = config.savePath();
-    if (directory.isEmpty() || !QDir(directory).exists() ||
-        !QFileInfo(directory).isWritable()) {
-        directory =
-          QStandardPaths::writableLocation(QStandardPaths::PicturesLocation);
+    QFileInfo info(path);
+    if (info.isDir()) {
+        return QDir(QDir(path).absolutePath() + "/" + parsedPattern() + "." +
```

Comment:
Is using the "/" in the file path valid for Windows?

Reply:
QDir should convert the delimiter automatically.